prime-router/src/main/kotlin/tokens/Jwk.kt

```diff
@@ -193,5 +193,18 @@ data class JwkSet(
 
             return nullSafeJwkSets.filter { jwkSet -> jwkSet.scope != scope } + listOf(updatedJwkSet)
         }
+
+        fun isValidKidForSet(jwkSets: List<JwkSet>?, scope: String, kid: String?): Boolean {
```

Comment:
As it currently stands, this has not been enforced and their are senders with the same `kid` for different keys.  This check will only run for new keys getting added and the migration plan we'll need to account for this.